src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java

```diff
@@ -73,80 +74,68 @@
+        UrlEncodedFormEntity entity = new UrlEncodedFormEntity(Arrays.asList(pairs));
+
+        int proxyPort = config.getInt(Config.PROXY_PORT);
+        String proxyHostName = config.getString(Config.PROXY_HOST);
+        String proxyUser = config.getString(Config.PROXY_USERNAME);
+        String proxyPassword = config.getString(Config.PROXY_PASSWORD);
+        String ntlmDomain = config.getString(Config.PROXY_NTLM_DOMAIN);
+        proxyHostName = proxyHostName != null ? proxyHostName.trim() : "";
+        proxyUser = proxyUser != null ? proxyUser.trim() : "";
+        proxyPassword = proxyPassword != null ? proxyPassword.trim() : "";
+        ntlmDomain = ntlmDomain != null ? ntlmDomain.trim() : "";
+
+        post.addHeader("User-Agent", VersionInfo.info());
+        post.setEntity(entity);
+
+        //proxy
+        if (proxyHostName.length() > 0) {
+            InetSocketAddress proxyAddress = new InetSocketAddress(proxyHostName, proxyPort);
+            HttpHost proxyHost = new HttpHost(proxyAddress.getHostName(), proxyAddress.getPort(), "http");
+            AuthScope scope = new AuthScope(proxyAddress.getHostName(), proxyAddress.getPort(), null, null);
+            Credentials credentials = new UsernamePasswordCredentials(proxyUser, proxyPassword);
+
+            if (ntlmDomain.length() > 0) {
+                credentials = new NTCredentials(proxyUser, proxyPassword, InetAddress.getLocalHost().getCanonicalHostName(), ntlmDomain);
+            }
 
-                RequestConfig.Builder requestConfigBuilder = RequestConfig.custom();
-                requestConfigBuilder.setProxy(proxyHost);
-                post.setConfig(requestConfigBuilder.build());
+            RequestConfig.Builder requestConfigBuilder = RequestConfig.custom();
+            requestConfigBuilder.setProxy(proxyHost);
+            post.setConfig(requestConfigBuilder.build());
 
-                CredentialsProvider credentialsprovider = new BasicCredentialsProvider();
-                credentialsprovider.setCredentials(scope, credentials);
-                client = HttpClientBuilder.create().setDefaultCredentialsProvider(credentialsprovider).build();
-            }
-            CloseableHttpResponse response = null;
-            try {
-                if (ntlmDomain.length() > 0) {
-                    // need to send a HEAD request to trigger NTLM authentication
-                    HttpHead head = new HttpHead("http://salesforce.com");
-                    try {
-                        response = client.execute(head);
-                    } finally {
-                        response.close();
-                    }
+            CredentialsProvider credentialsprovider = new BasicCredentialsProvider();
+            credentialsprovider.setCredentials(scope, credentials);
+            httpClientBuilder.setDefaultCredentialsProvider(credentialsprovider).build();
+        }
+
+        try (CloseableHttpClient httpClient = httpClientBuilder.build()) {
```

Comment:
Nice. Please make sure to run the integration tests before we pull in the change.